nrf_wifi/README.rst

```diff
@@ -3,9 +3,196 @@
+     - Availability of the per-AC and global tokens.
+#. If the frame can be transmitted, **UMAC IF** dequeues the frame from the transmit pending queue and enqueues it to the transmit queue.
+#. **UMAC IF** processes the transmit queue:
+   * It prepares the necessary information for **UMAC** firmware in the form of a ``struct nrf_wifi_tx_buff`` command.
+   * It `copies` (this can be a memory copy or a DMA depending on underlying **BAL** implementation) each frame to the appropriate memory location in the Packet RAM
+#. **UMAC IF** sends the ``struct nrf_wifi_tx_buff`` command to the **HAL** layer using the typical command flow, as described in the previous section.
+
+Data flow: Receive
+==================
+
+The key difference between the transmit and receive flow is that the RX packet buffers need to be pre-programmed to the nRF70 Series Wi-Fi chipset as packets can be received at any time.
+Unlike the transmit flow, the receive flow does not have a token bucket mechanism to control the flow of packets.
+
+Queues
+------
+
+The nRF70 Series Wi-Fi driver does not implement any queues for the receive path.
+It is a circular buffer with a configurable size :kconfig:option:`CONFIG_NRF700X_RX_NUM_BUFS`.
+
+Flow: initialization
+--------------------
+
+The nRF70 Series Wi-Fi driver implements the following flow during initialization:
+
+1. **FMAC** allocates a set of buffers in the Packet RAM for the receive path.
+    The number of buffers is controlled by the :kconfig:option:`CONFIG_NRF700X_RX_NUM_BUFS` Kconfig option.
+#. **FMAC** prepares a ``struct nrf_wifi_rx_buff`` command and sends it to the **HAL** layer.
+#. **HAL** sends the ``struct nrf_wifi_rx_buff`` command to the **UMAC** firmware using the typical command flow, as described in the previous section.
+
+Flow: Receive
+-------------
+
+The nRF70 Series Wi-Fi driver implements the following flow for receiving a frame:
+
+1. **UMAC** firmware receives a frame from the air, processes it and stores it in the Packet RAM.
+    Prepares ``struct nrf_wifi_rx_buff`` and generates an interrupt to the host processor.
+#. **UMAC IF** receives the interrupt, processes ``struct nrf_wifi_rx_buff`` event, and copies the frame from the Packet RAM to the receive buffer.
+#. **UMAC IF** performs 802.11 decapsulation and passes the frame to the networking stack.
+
+Flow: Refill
+-------------
+
+The nRF70 Series Wi-Fi driver implements a refill mechanism to ensure that the Packet RAM is always filled with receive buffers.
+
+1. **UMAC IF** allocates a new buffer for the received frame.
+#. **UMAC IF** prepares a ``struct nrf_wifi_rx_buff`` command and sends it to the **HAL** layer.
+#. **HAL** sends the ``struct nrf_wifi_rx_buff`` command to the **UMAC** firmware using the typical command flow, as described in the previous section.
+
+
 .. toctree::
    :maxdepth: 1
    :caption: Modules:
```

Comment:
We cannot have a page with both the content divided in sections and a toctree of pages at the bottom. This will mess up how the hierarchy is displayed in the documentation and lead to a situation where you cannot distinguish between sections and subpages. All the new content on this page needs to be moved to a new, separate page.